core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -1155,7 +1155,9 @@ boolean scheduleTask(Workflow workflow, List<Task> tasks) {
                 if (workflowSystemTask == null) {
                     throw new ApplicationException(NOT_FOUND, "No system task found by name " + task.getTaskType());
                 }
-                task.setStartTime(System.currentTimeMillis());
+                if (task.getStatus() != null && task.getStatus().isTerminal() && task.getStartTime() == 0) {
```

Comment:
This should be `!task.getStatus().isTerminal()`

Reply:
Fixed